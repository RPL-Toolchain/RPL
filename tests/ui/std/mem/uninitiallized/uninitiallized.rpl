pattern std_mem_uninitiallized 

patt {
    uninitialized[
        $T: ty
    ] = pub fn _ (..) -> _ {
        let $uninit: $T = std::mem::uninitialized();
    }
}