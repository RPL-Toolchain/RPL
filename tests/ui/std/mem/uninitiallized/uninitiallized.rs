//@ignore-on-host
use std::mem::uninitialized;
use std::mem::MaybeUninit;

#[rpl::dump_mir()]
fn uninitialized_value<T>() -> T {
    unsafe { uninitialized() }
}