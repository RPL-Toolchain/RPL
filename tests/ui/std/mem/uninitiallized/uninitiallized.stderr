note: MIR of `uninitialized_value`
  --> tests/ui/std/mem/uninitiallized/uninitiallized.rs:6:1
   |
LL |   #[rpl::dump_mir()]
   |   ------------------ MIR dumpped because of this attribute
LL | / fn uninitialized_value<T>() -> T {
LL | |     unsafe { uninitialized() }
LL | | }
   | |_^
   |
   = note: see `/Users/stuuupidcat/home/code/projects/RPL/mir_dump/uninitiallized.uninitialized_value.-------.dump_mir..mir` for dumpped MIR
note: locals and scopes in this MIR
  --> tests/ui/std/mem/uninitiallized/uninitiallized.rs:6:1
   |
LL |   fn uninitialized_value<T>() -> T {
   |   ^                              - _0: T; // scope[0]
   |  _|
   | |
LL | |     unsafe { uninitialized() }
LL | | }
   | |_^ scope[0]
note: bb0: {
          _0 = std::mem::uninitialized::<T>() -> [return: bb1, unwind continue]; // scope[0]
      }
  --> tests/ui/std/mem/uninitiallized/uninitiallized.rs:7:14
   |
LL |     unsafe { uninitialized() }
   |              ^^^^^^^^^^^^^^^ _0 = std::mem::uninitialized::<T>() -> [return: bb1, unwind continue]; // scope[0]
note: bb1: {
          return; // scope[0]
      }
  --> tests/ui/std/mem/uninitiallized/uninitiallized.rs:8:2
   |
LL | }
   |  ^ return; // scope[0]

error: abort due to debugging
  --> tests/ui/std/mem/uninitiallized/uninitiallized.rs:5:1
   |
LL | #[rpl::dump_mir()]
   | ^^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: aborting due to 1 previous error

